graph/src/components/subgraph/instance.rs

```diff
@@ -76,6 +88,222 @@ pub struct BlockState {
 
     // Marks whether a handler is currently executing.
     in_handler: bool,
+
+    pub metrics: BlockStateMetrics,
+}
+
+#[derive(Debug)]
+pub struct BlockStateMetrics {
```

Comment:
I'd put this in a submodule in a separate file, just to keep things organized. But up to you.

Reply:
done